testutil/src/main/java/org/spine3/test/NullToleranceTest.java

```diff
@@ -0,0 +1,500 @@
+         */
+        private boolean hasExpectedStackTrace(Throwable cause) {
+            final StackTraceElement[] stackTraceElements = cause.getStackTrace();
+            final StackTraceElement preconditionsElement = stackTraceElements[0];
+            final boolean preconditionClass = Preconditions.class.getName()
+                                                                 .equals(preconditionsElement.getClassName());
+            if (!preconditionClass) {
+                return false;
+            }
+
+            final StackTraceElement targetClassElement = stackTraceElements[1];
+            final String targetMethodName = targetClassElement.getMethodName();
+            final boolean correct = method.getName()
+                                          .equals(targetMethodName);
+            if (!correct) {
+                return false;
+            }
+
+            final boolean correctClass = targetClassName.equals(targetClassElement.getClassName());
+            return correctClass;
+        }
+    }
+
+    /**
+     * Provides the default values for the method arguments.
+     */
+    private static class DefaultValuesProvider {
+
+        private static final Class[] EMPTY_PARAMETER_TYPES = {};
+        private static final Object[] EMPTY_ARGUMENTS = {};
+        private static final String STRING_DEFAULT_VALUE = "";
+        private static final String METHOD_NAME = "getDefaultInstance";
+        private final Map<?, ?> defaultValues;
+
+        private DefaultValuesProvider(Map<?, ?> defaultValues) {
+            this.defaultValues = defaultValues;
+        }
+
+        /**
+         * Returns the default value for the method argument by the method argument type.
+         * If default value does not provide, throws {@link IllegalStateException} otherwise.
+         *
+         * @param key the {@code Class} of the method argument
+         * @return the default value
+         */
+        private Object getDefaultValue(Class<?> key) {
+            Object result = defaultValues.get(key);
+            if (result != null) {
+                return result;
+            }
+
+            final boolean primitive = allPrimitiveTypes().contains(key);
+            if (primitive) {
+                result = defaultValue(key);
+            }
+
+            final boolean wrapper = isWrapperType(key);
+            if (result == null && wrapper) {
+                final Class<?> unwrappedPrimitive = unwrap(key);
+                result = defaultValue(unwrappedPrimitive);
+            }
+
+            final Class<GeneratedMessageV3> messageClass = GeneratedMessageV3.class;
+            final boolean message = messageClass.isAssignableFrom(key);
+            if (result == null && message) {
+                result = getDefaultMessageInstance(key);
+            }
+
+            final Class<String> stringClass = String.class;
+            if (result == null && stringClass.isAssignableFrom(key)) {
```

Comment:
Why not populate the default value for `String` in the same way as you do for `Number`s?